frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/PipelineRunTaskRunGraph.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { ChartVoronoiContainer } from '@patternfly/react-charts';
+import { GraphEmpty } from '@console/internal/components/graphs/graph-empty';
+import { formatDuration } from '@console/internal/components/utils/datetime';
+import { LoadingInline } from '@console/internal/components/utils';
+import { usePipelineRunTaskRunPoll } from '../hooks';
+import {
+  PipelineMetricsGraphProps,
+  getRangeVectorData,
+  getYaxisValues,
+} from './pipeline-metrics-utils';
+import { LineChart } from './charts/lineChart';
+import { DEFAULT_CHART_HEIGHT } from '../const';
+
+import './PipelineChart.scss';
+
+const PipelineRunTaskRunGraph: React.FC<PipelineMetricsGraphProps> = ({
+  pipeline,
+  timespan,
+  interval,
+  width = 650,
+}) => {
+  const {
+    metadata: { name, namespace },
+  } = pipeline;
+
+  const [runData, runDataError, runDataLoading] = usePipelineRunTaskRunPoll({
+    name,
+    namespace,
+    timespan,
+    delay: interval,
+  });
+
+  if (runDataLoading) {
+    return <LoadingInline />;
+  }
+
+  const pipelineRunResultData = runData?.data?.result ?? [];
+  if (runDataError || pipelineRunResultData.length === 0) {
+    return <GraphEmpty height={DEFAULT_CHART_HEIGHT} />;
+  }
+
+  const pRuns =
+    getRangeVectorData(runData, (r) => {
+      return r.metric.pipelinerun;
+    }) ?? [];
+  const finalObj: { [x: string]: { x: string; y: number }[] } = pRuns.reduce((acc, prun) => {
+    if (!prun) return acc;
+
+    const obj = prun[prun.length - 1];
```

Comment:
For every loop use the last one? This loop doesn't do anything different each loop, does it?

Reply:
For every loop, finding the last element in the values array and forming  an obj based on it.

Reply:
Doh - this is what I get for doing reviews late at night.